html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication.pm

```diff
@@ -219,30 +223,27 @@ sub create_local_account {
 
     # We create a "password" (also known as a user account) using the pid
     # with different parameters coming from the authentication source (ie.: expiration date)
-    my $actions = &pf::authentication::match( $self->source->id, $auth_params );
-
-    # We push an unregistration date that was previously calculated (setUnRegDate) that handle dynamic unregistration date and access duration
-    my $action = pf::Authentication::Action->new({
-        type    => $Actions::SET_UNREG_DATE, 
-        value   => $self->new_node_info->{'unregdate'},
-        class   => pf::Authentication::Action->getRuleClassForAction($Actions::SET_UNREG_DATE),
-    });
-    # Hack alert: We may already have a "SET_UNREG_DATE" action in the array and since the way the authentication framework is working is by going
-    # through the actions on a first hit match, we want to make sure the unregistration date we computed (because we are taking care of the access duration,
-    # dynamic date, ...) will be the first in the actions array.
-    unshift (@$actions, $action);
+    $actions = $actions // &pf::authentication::match( $self->source->id, $auth_params );
 
-    $password = pf::password::generate($self->app->session->{username}, $actions, $password);
+    my $login_amount = ($self->source->local_account_logins eq "0") ? undef : $self->source->local_account_logins;
```

Comment:
I'd use a constant (constant is the key :tm: ) for the '0' I hate to grep for '0' if you know what I mean ;)
lib/pf/Authentication/CreateLocalAccountRole.pm